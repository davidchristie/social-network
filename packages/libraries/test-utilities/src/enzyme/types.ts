import { ReactWrapper, ShallowWrapper } from "enzyme";

export type Wrapper = ReactWrapper | ShallowWrapper;
