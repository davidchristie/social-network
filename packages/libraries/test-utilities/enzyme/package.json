{
  "main": "../dist/enzyme/index.js",
  "name": "test-utilities/enzyme",
  "private": true,
  "version": "1.0.0"
}
