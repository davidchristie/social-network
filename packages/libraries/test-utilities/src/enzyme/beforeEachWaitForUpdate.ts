import { Wrapper } from "./types";

export default function beforeEachWaitForUpdate (getWrapper: () => Wrapper) {
  beforeEach(async () => {
    await new Promise(resolve => window.setTimeout(resolve, 0));
    getWrapper().update();
  });
}
