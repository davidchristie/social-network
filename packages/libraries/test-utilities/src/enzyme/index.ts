export { default as beforeEachWaitForUpdate } from "./beforeEachWaitForUpdate";
export { default as itContainsComponent } from "./itContainsComponent";
