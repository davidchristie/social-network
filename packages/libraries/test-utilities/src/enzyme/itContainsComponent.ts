import React from "react";
import { Wrapper } from "./types";

export default function itContainsComponent (
  name: string,
  getWrapper: () => Wrapper,
  Component: React.ComponentType
) {
  it(name, () => {
    expect(getWrapper().find(Component).exists()).toBe(true);
  });
}
