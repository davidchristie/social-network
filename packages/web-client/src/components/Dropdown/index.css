.Dropdown {
  background: white;
  border: 1px solid black;
  box-shadow: 4px 4px 5px rgba(0, 0, 0, 0.4);
  padding: 1em;
  position: absolute;
  right: 0;
  top: 100%;
  z-index: 2;
}

.Dropdown > * {
  width: 100%;
}
