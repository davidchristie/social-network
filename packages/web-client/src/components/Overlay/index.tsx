import React from "react";

import "./index.css";

interface Props {
  onClick?: (event: React.MouseEvent<HTMLDivElement>) => void;
}

export default class Overlay extends React.Component<Props> {
  public render () {
    return (
      <div
        className="Overlay"
        onClick={this.props.onClick}
      />
    );
  }
}
