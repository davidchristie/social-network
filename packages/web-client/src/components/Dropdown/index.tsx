import React from "react";

import Overlay from "../Overlay";
import "./index.css";

interface Props {
  onClose?: () => void;
  open: boolean;
}

export default class Dropdown extends React.Component<Props> {
  public render () {
    if (!this.props.open) {
      return null;
    }
    return (
      <React.Fragment>
        <Overlay onClick={this.props.onClose} />
        <div className="Dropdown">
          {this.props.children}
        </div>
      </React.Fragment>
    );
  }
}
