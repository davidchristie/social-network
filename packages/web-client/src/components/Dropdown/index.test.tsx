import { shallow, ShallowWrapper } from "enzyme";
import React from "react";

import Overlay from "../Overlay";
import Dropdown from "./index";

describe("Dropdown component", () => {
  let wrapper: ShallowWrapper;

  describe("when closed", () => {
    beforeEach(() => {
      wrapper = shallow(<Dropdown open={false} />);
    });

    it("matches snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("when open", () => {
    const onClose = jest.fn();

    beforeEach(() => {
      wrapper = shallow(<Dropdown onClose={onClose} open={true} />);
    });

    it("renders overlay", () => {
      expect(wrapper.find(Overlay).exists()).toBe(true);
    });

    it("is closed by clicking overlay", () => {
      wrapper.find(Overlay).simulate("click");
      expect(onClose).toHaveBeenCalledTimes(1);
    });

    it("matches snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
