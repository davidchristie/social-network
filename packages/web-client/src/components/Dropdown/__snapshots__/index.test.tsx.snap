// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown component when closed matches snapshot 1`] = `""`;

exports[`Dropdown component when open matches snapshot 1`] = `
<Fragment>
  <Overlay
    onClick={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
  />
  <div
    className="Dropdown"
  />
</Fragment>
`;
